fuels-core/src/lib.rs

```diff
@@ -44,15 +44,29 @@ impl Default for ParamType {
 }
 
 impl ParamType {
-    // Checks whether the `ParamType` is bigger than a `WORD`
+    // Checks if the `ParamType` is bigger than a `WORD`.
+    // This is important because, depending on whether its
+    // bigger or smaller than a `WORD`, the returned data
+    // will be inside a `ReturnData` receipt or a `Return` receipt.
     pub fn bigger_than_word(&self) -> bool {
-        match *self {
+        match &*self {
+            // Bits256 Always bigger than one `WORD`.
             Self::B256 => true,
-            Self::String(size) => size > 8,
+            // Strings are bigger than one `WORD` when its size > 8.
+            Self::String(size) => size > &8,
+            Self::Struct(params) => match params.len() {
+                // If only one component in this struct
+                // check if this element itself is bigger than a `WORD`.
+                1 => params[0].bigger_than_word(),
+                _ => true,
+            },
+            // Enums are always in `ReturnData`.
+            Self::Enum(_params) => true,
+            // Arrays seem to always be inside `ReturnData`.
```

Comment:
Is this expected or a bug? cc @otrho

Reply:
I'm guessing at this stage it's always true since even an enum with a single variant will still have a tag and a value, even if it's just `unit`.  So it can't fit in a register.  After we add some gnarly optimisations this may no longer be true, but only if we want to go there.

Reply:
Oh I was pointing at the ``Arrays seem to always be inside `ReturnData`.``, not the enums 😂

Reply:
Unfortunately I had to do a lot of guesswork to get this working and figure out where each returned type would be located at. 

As for arrays, in the tests I did, they were always in ReturnData (there are few new array-as-output tests added in this PR). But then again, I didn't fuzz or anything, just tested a few cases and focused on solving some existing blockers. 

But, intuitively (which could be wrong of course), if any primitive type takes a whole word. Even if it doesn't fill it all we pad it. So an array should be larger than a word? Maybe unless is an array of size zero or one? 😅

Reply:
An array of size 0 or 1 should only take up a single word. Could you add a test for that @digorithm?